library/src/test/java/com/novoda/downloadmanager/lib/CriteriaTest.java

```diff
@@ -0,0 +1,180 @@
+                .build();
+
+        CriteriaAssert.assertThat(criteria)
+                .hasSelection(SELECTION_COLUMN_AND_ANOTHER_SELECTION_COLUMN_QUERY)
+                .hasArguments(new String[]{ARGUMENT, ARGUMENT});
+    }
+
+    @Test
+    public void givenTwoSelectionsWhenWeBuildWithOrThenTheSelectionQueryIsCorrect() {
+        Criteria criteria = new Criteria.Builder()
+                .withSelection(SELECTION_COLUMN, Criteria.Wildcard.EQUALS)
+                .withArgument(ARGUMENT)
+                .or()
+                .withSelection(ANOTHER_SELECTION_COLUMN, Criteria.Wildcard.EQUALS)
+                .withArgument(ARGUMENT)
+                .build();
+
+        CriteriaAssert.assertThat(criteria)
+                .hasSelection(SELECTION_COLUMN_OR_ANOTHER_SELECTION_COLUMN_QUERY)
+                .hasArguments(new String[]{ARGUMENT, ARGUMENT});
+    }
+
+    @Test
+    public void givenMultipleSelectionsWhenWeBuildThenTheSelectionQueryIsCorrect() {
+        Criteria criteria = new Criteria.Builder()
+                .withSelection(SELECTION_COLUMN, Criteria.Wildcard.EQUALS)
+                .withArgument(ARGUMENT)
+                .or()
+                .withSelection(ANOTHER_SELECTION_COLUMN, Criteria.Wildcard.EQUALS)
+                .withArgument(ARGUMENT)
+                .and()
+                .withSelection(SELECTION_COLUMN, Criteria.Wildcard.EQUALS)
+                .withArgument(ARGUMENT)
+                .build();
+
+        CriteriaAssert.assertThat(criteria)
+                .hasSelection(SELECTION_COLUMN_OR_ANOTHER_SELECTION_COLUMN_QUERY + AND + SELECTION_QUERY)
+                .hasArguments(new String[]{ARGUMENT, ARGUMENT, ARGUMENT});
+    }
+
+    @Test
+    public void givenMultipleSelectionsWithInnerCriteriaWhenWeBuildThenTheSelectionQueryIsCorrect() {
+        Criteria criteria = new Criteria.Builder()
+                .withSelection(SELECTION_COLUMN, Criteria.Wildcard.EQUALS)
+                .withArgument(ARGUMENT)
+                .or()
+                .withInnerCriteria(
+                        new Criteria.Builder()
+                                .withSelection(ANOTHER_SELECTION_COLUMN, Criteria.Wildcard.EQUALS)
+                                .withArgument(ARGUMENT)
+                                .and()
+                                .withSelection(SELECTION_COLUMN, Criteria.Wildcard.EQUALS)
+                                .withArgument(ARGUMENT)
+                                .build())
+                .build();
+
+        CriteriaAssert.assertThat(criteria)
+                .hasSelection(SELECTION_QUERY + OR + "(" + ANOTHER_SELECTION_COLUMN_QUERY + AND + SELECTION_QUERY + ")")
+                .hasArguments(new String[]{ARGUMENT, ARGUMENT, ARGUMENT});
+    }
+
+    @Test
+    public void givenMultipleSelectionsWithMultipleInnerCriteriaWhenBuildingThenTheSelectionQueryIsCorrect() {
+        Criteria criteria = new Criteria.Builder()
+                .withInnerCriteria(
```

Comment:
https://github.com/novoda/download-manager/pull/79/files#diff-5fabc9dc335ace9af378b7c1c777a861R179 here we are using it. That `AND` needs to be inner (or in brackets) in order to take precedence in evaluation in between the `OR`s.